### DISCUSSION TITLE
ninjabackup often fails to copy chyrp's sqlitedb

### DISCUSSION THREAD

**Discussion by roughnecks** [xenocrat/chyrp-lite#269] [Q&A] [answered]
Hello,

this is not strictly related to chyrp-lite, but quite.
I have several sqlite dbs which I back up at night with ninjabackup and quite often the backup fails (sends me an email) with the following error, always about chyrp:

/bin/tar: /var/lib/xmpp-it/xmpp-it.db: file changed as we read it

Since I'm running this at night and we are just a little community, can it be due to bots accessing the site? Why the db would change though if the site is accessed in read-only?
Should I stop nginx a couple minutes before the backup starts or is there anything else I might try?

Thanks

**Comment by xenocrat** [comment, marked as answer]
Hello there,

Sessions are also stored in the database. Anyone accessing the site will cause a session cookie to be created and stored in the database.

[Self-identified bots](https://github.com/xenocrat/chyrp-lite/blob/master/includes/common.php#L76-L81) are denied session cookies, but unfortunately a lot of bots do not identify honestly.

The easiest solution, if possible, is to have your backup script create the file _includes/upgrading.lock_, do the backup, and then delete that file immediately afterwards. This will throw a 503 Service Unavailable message to all visitors and prevent all database and configuration file activity while your backup is in progress.

**Comment by roughnecks** [comment]
Thanks, that should be easily done!
